emukit/examples/dynamic_negative_lower_confidence_bound/dnlcb.py

```diff
@@ -0,0 +1,28 @@
+from emukit.bayesian_optimization.acquisitions import NegativeLowerConfidenceBound
+from emukit.core.interfaces import IModel, IDifferentiable
+import numpy as np
+from typing import Union
+
+class DynamicNegativeLowerConfidenceBound(NegativeLowerConfidenceBound):
+    def __init__(self, model: Union[IModel, IDifferentiable], input_space_size: int, delta: float) -> None:
+        """
+        Dynamic extension of the LCB acquisition. The beta coefficient is updated at each iteration, based on the explorativeness parameter delta which is inversely
+        proportional to beta itself - the higher the delta the less explorative the selection will be.
+        Please consider that regret bounds based on the dynamic exploration coefficient only hold for selected kernel classes exhibiting boundedness and smoothness.
+        See the base class for references.
+        This class may also be taken as a reference for acquisition functions that dynamically update their parameters thanks to the update_parameters() hook.
+        :param model: The underlying model that provides the predictive mean and variance for the given test points
+        :param input_space_size: the size of the finite D grid on which the function is evaluated
+        :param delta: the exploration parameter determining the beta exploration coefficient; delta must be in (0, 1) and it is inversely related to beta
+        """
+        assert input_space_size > 0, "Invalid dimension provided"
+        assert 0 < delta < 1, "Delta must be in (0, 1)"
```

Comment:
Within examples using asserts is probably fine, but in the library we normally raise exceptions. It's a better practice overall: they are easier to catch, can be inhereted and documented, and don't look like debugging in the code.

Reply:
Ok, I've left it as it is now since it's just an example, but you may return a `ValueException` if you like of course.